interview-questions.md

```diff
@@ -16,53 +16,60 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
 1. What is git? What is the difference between git and Github?
 
   Your answer:
-
-  Researched answer:
-
+Git is a software called Global Interface Technology used in the Github site. Github is the website using the git software.
+  Researched answer: The git software is used mainly for tracking changes in files. The git software helps progammers coordinate work when they are collaborating to develop source codes.
+      The Github is a repository that is cloud based, uses git software that hosts files or project for developers.
 
 
 2. Which JavaScript operators will return a Boolean value?
 
   Your answer:
 
-  Researched answer:
+The following operators will return a Boolean value:
 
+    =  - the equals sign
+    <  - the less than sign
+    >  - the more than sign
+  Researched answer: In addition to the above operators, the following operators will return Boolean value:
+          * &&  - And relational operator
+          * ||  - Not, another relational operator
+          
+          Sometimes these operators are also used in combination with one another to further narrow down some conditions.
 
 
 3. What is an index? What is the difference between index and value?
 
   Your answer:
+An index in an array tells us the location, like an address. The index will tell us the location, while the value tells us what is in that location. If If I will equate it to a mail address, the value is like the name that resides in it while the index is the address which points to the exact location.
+  Researched answer: In an array, an index is a specific location where a value is stored. Value is the item of value stored  in the location pointed to by the index. A value must be a data type that the Javascript accepts.
 
-  Researched answer:
 
 
+4. What is iteration? 
+  Your answer: Doing a set of operations or functions multiple times.
 
-4. What is iteration?
-
-  Your answer:
-
-  Researched answer:
+  Researched answer: Repeating execution of codes until set stopping condition is met. It is very usefull in handling large data. These large amounts of data are usually in string or arrays.
```

Comment:
Definitely more to this and it goes quite deep! I have been asked this question in interviews before.